tests/Shared/WorkloadTesting/AspireProject.cs

```diff
@@ -199,6 +286,51 @@ public async Task BuildAsync(string[]? extraBuildArgs = default, CancellationTok
         res.EnsureSuccessful();
     }
 
+    public async Task<IPage> OpenDashboardPageAsync(IBrowserContext context, int timeoutSecs = DashboardAvailabilityTimeoutSecs)
+    {
+        string dashboardUrlToUse;
+        if (Environment.GetEnvironmentVariable("DASHBOARD_URL_FOR_TEST") is string dashboardUrlForTest)
+        {
+            dashboardUrlToUse = dashboardUrlForTest;
+        }
+        else
+        {
+            dashboardUrlToUse = DashboardUrl!;
+        }
+        if (string.IsNullOrEmpty(dashboardUrlToUse))
+        {
+            throw new InvalidOperationException("Dashboard URL is not available");
+        }
+
+        CancellationTokenSource cts = new();
+        cts.CancelAfter(TimeSpan.FromSeconds(DashboardAvailabilityTimeoutSecs));
+        await WaitForDashboardToBeAvailableAsync(dashboardUrlToUse, _testOutput, cts.Token).ConfigureAwait(false);
+
+        var dashboardPage = await context.NewPageWithLoggingAsync(_testOutput);
+        await dashboardPage.GotoAsync(dashboardUrlToUse);
+
+        return dashboardPage;
+    }
+
+    public Task WaitForDashboardToBeAvailableAsync(CancellationToken token = default)
```

Comment:
```suggestion
    public Task WaitForDashboardToBeAvailableAsync(CancellationToken cancellationToken = default)
```

This is a better name than `token`.